src/k8s/pkg/k8sd/controllers/csrsigning/validate.go

```diff
@@ -9,19 +12,30 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-func validateCSR(obj *certv1.CertificateSigningRequest) error {
+// validateCSR checks the CSR for common requirements and returns an error if it fails.
+// validateCSR is used to validate CSRs that are auto-approved.
```

Comment:
better to describe what the function _does_, not what are callers will use it for.

Reply:
fair point - fixed